src/aiperf/ui/dashboard/realtime_telemetry_dashboard.py

```diff
@@ -0,0 +1,312 @@
+        )
+
+        return [
+            Text(metric_name, style="bold cyan", justify="left"),
+            *[
+                self._format_value(getattr(metric, field))
+                for field in self.STATS_FIELDS
+            ],
+        ]
+
+    def _format_value(self, value) -> Text:
+        """Format a metric value for display (matches console exporter format)."""
+        if value is None:
+            return Text("N/A", justify="right", style="dim")
+
+        if not isinstance(value, int | float):
+            return Text(str(value), justify="right", style="green")
+
+        value_str = f"{value:.2e}" if abs(value) >= 1000000 else f"{value:,.2f}"
+
+        return Text(value_str, justify="right", style="green")
+
+
+class SingleNodeView(VerticalScroll):
+    """Display all GPUs for a single node directly."""
+
+    DEFAULT_CSS = """
+    SingleNodeView {
+        height: 100%;
+        padding: 1 2;
+    }
+    SingleNodeView > Static {
+        background: $panel;
+        padding: 0 1;
+        margin: 0 0 1 0;
+        text-style: bold;
+    }
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.gpu_tables: dict[str, GPUMetricsTable] = {}
+
+    def compose(self) -> ComposeResult:
+        """Compose single node view (no initial widgets - GPU tables added dynamically)."""
+        yield from ()
+
+    def update(self, metrics: list[MetricResult]) -> None:
+        """Update display with all GPUs from all nodes."""
+
+        gpus = self._group_metrics_by_gpu(metrics)
+
+        for gpu_key, gpu_metrics in gpus.items():
+            if gpu_key not in self.gpu_tables:
+                if not self.is_mounted:
+                    continue
+
+                endpoint, gpu_index, gpu_uuid, model_name = self._extract_gpu_info(
+                    gpu_metrics[0]
+                )
+                gpu_table = GPUMetricsTable(endpoint, gpu_uuid, gpu_index, model_name)
+                self.gpu_tables[gpu_key] = gpu_table
+                self.mount(gpu_table)
+
+            self.gpu_tables[gpu_key].update(gpu_metrics)
+
+    def _group_metrics_by_gpu(
+        self, metrics: list[MetricResult]
+    ) -> dict[str, list[MetricResult]]:
+        """Group metrics by GPU."""
+        gpus = {}
+        for metric in metrics:
+            gpu_key = self._extract_gpu_key_from_tag(metric.tag)
+            if gpu_key not in gpus:
+                gpus[gpu_key] = []
+            gpus[gpu_key].append(metric)
+        return gpus
+
+    def _extract_gpu_key_from_tag(self, tag: str) -> str:
+        """Extract GPU identifier from metric tag including endpoint info.
+
+        Tag format: metric_name_dcgm_http___localhost_9400_metrics_gpu0_uuid
+        Returns: dcgm_http___localhost_9400_metrics_0_uuid
+        """
+        if "_dcgm_" not in tag or "_gpu" not in tag:
+            return "unknown"
+
+        dcgm_and_gpu = tag.split("_dcgm_")[1]
+        return dcgm_and_gpu.replace("_gpu", "_")
```

Comment:
yeah i think a lot of this can be simplified if the realtime telemetry returns the structurted data instead of list[MetricResult]